front_end/helper.py

```diff
@@ -56,7 +56,13 @@ def convert_markdown_to_html(text):
         return ""
 
     markdown = Markdown()
-    html = re.sub(r"youtube:([-_a-zA-Z0-9]+)", r"<iframe width='800' height='550' src='https://www.youtube.com/embed/\1?controls=1'></iframe>", text)
+
+    html = re.sub(r"<(/*)span(.*?)>", "", text) # Removes opening and closing <span> tags.
+    html = re.sub(r"<(div|br)(.*?)>", "\n", html) # Replaces <br> and <div> tags with a newline.
+    html = re.sub(r"<(/*)div(.*?)>", "", html) # Removes closing </div> tags.
+    html = re.sub(r"youtube:([-_a-zA-Z0-9]+)", r"<iframe width='800' height='550' src='https://www.youtube.com/embed/\1?controls=1'></iframe>", html)
+    html = re.sub(r'<img src="([^">]+?)">', r"![](\1)", html)
+
```

Comment:
When we added functionality to paste images directly into a course/assignment/exercise introduction, it required using ```contenteditable divs``` rather than ```textareas```, leading to some extra html markup getting saved in ```course_details["introduction"]```. These regexes remove said html markup before converting markdown to html.